cli/flox-watchdog/src/process.rs

```diff
@@ -193,9 +198,17 @@ impl PidWatcher {
 impl Watcher for PidWatcher {
     fn wait_for_termination(&mut self) -> Result<WaitResult, Error> {
         loop {
-            self.update_watchlist()?;
+            self.update_watchlist(false)?;
             if self.should_clean_up()? {
-                return Ok(WaitResult::CleanUp);
+                // Don't hold the lock during normal polling to avoid contention
+                // But when we're actually ready to cleanup, we need to hold the lock
+                // TODO: could probably refactor and get rid of the unwrap
+                let locked_activations = self.update_watchlist(true)?.ok_or(anyhow::anyhow!(
```

Comment:
Nit: seeing a bare `true` or `false` when the method doesn't start with `should_*` is a little ambiguous. One way around this is to have a private method that takes the bool and have wrapper methods like `update_watchlist_while_holding_lock` in the public API.

Reply:
Yeah I could add an additional method to the trait. I'd kinda prefer just dropping the trait since we aren't using it and it's annoying to have to change